source/MaterialXGenShader/ShaderNode.h

```diff
@@ -144,9 +144,16 @@ class ShaderInput : public ShaderPort
     /// Get optional channels value
     const string& getChannels() const { return _channels; }
 
+    /// Set isBindInput value. If true, either a BindInput or BindParam on a shaderref
+    void setIsBindInput(bool isBindInput) { _isBindInput = isBindInput; }
+
+    /// Get isBindInput value
+    const bool& getIsBindInput() const { return _isBindInput;  }
```

Comment:
I would recommend returning this boolean result by value, rather than by const reference.  Returning by const reference provides benefits for non-trivial types like std::string, but for a boolean it's clearer and equally efficient to return by value.